desc/protoparse/errors.go

```diff
@@ -92,15 +92,22 @@ type ErrorWithSourcePos struct {
 
 // Error implements the error interface
 func (e ErrorWithSourcePos) Error() string {
-	if e.Pos.Line <= 0 || e.Pos.Col <= 0 {
-		return fmt.Sprintf("%s: %v", e.Pos.Filename, e.Underlying)
+	sourcePos := e.GetPosition()
+	if sourcePos.Line <= 0 || sourcePos.Col <= 0 {
+		return fmt.Sprintf("%s: %v", sourcePos.Filename, e.Underlying)
 	}
-	return fmt.Sprintf("%s:%d:%d: %v", e.Pos.Filename, e.Pos.Line, e.Pos.Col, e.Underlying)
+	return fmt.Sprintf("%s:%d:%d: %v", sourcePos.Filename, sourcePos.Line, sourcePos.Col, e.Underlying)
 }
 
 // GetPosition implements the ErrorWithPos interface, supplying a location in
 // proto source that caused the error.
+//
+// If there is no underlying SourcePos, a new SourcePost with "<input>" as the
```

Comment:
If/when this ever happens, it's a material internal error. So I'm not sure it's valuable as part of the spec.

Reply:
Fair enough. I also had a typo, lol. Will fix

Reply:
Fixed